cpp/CMakeLists.txt

```diff
@@ -474,6 +486,16 @@ if(CUGRAPH_COMPILE_RAFT_LIB)
   endif()
 endif()
 
+unset(CUVS_LIB)
+if(CUGRAPH_COMPILE_CUVS AND CUGRAPH_USE_CUVS_STATIC)
+  set(CUVS_LIB cuvs::cuvs_static)
+elseif(CUGRAPH_COMPILE_CUVS)
+  set(CUVS_LIB cuvs::cuvs)
+else()
+  # When using prebuilt packages (CUGRAPH_COMPILE_CUVS=OFF), link to shared library
+  set(CUVS_LIB cuvs::cuvs)
+endif()
+
```

Comment:
```suggestion
set(CUVS_LIB cuvs::cuvs)
if(CUGRAPH_USE_CUVS_STATIC)
  set(CUVS_LIB cuvs::cuvs_static)
endif()

```

Reply:
Right. I believe this is the cause to the `overlinking` error reported in CI. Because of the ambiguity of my branches, it is calling `cuvs::cuvs` (second branch) when it should be calling `cuvs::cuvs_static` resulting in a shared binary dependency. By simplifying the logic and making `CUGRAPH_USE_CUVS_STATIC` the single source of truth for choosing static vs shared. this should resolve the issue and remove the ambiguity.